### PR TITLE
Add global_step for all loss function tasks

### LINKED ISSUE #458: add `global_step` variable as an argument for all loss function
Currently, only object detection model receives the variable `global_step` as the third argument. Consequently, in `lmnet/executor/train.py`, we need an if statement to pass `global_step` only for object detection model.

IMO, conditional branching by task type should be avoided whenever possible. To this end, we can pass `global_step` for all network definitions.

### PR DESCRIPTION
**Pull request by Joeper214**
## Motivation and Context

- fix #458 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature / Optimization (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by tkng** [comment]
I made a similar pull request a few days ago, but closed the pull request. (Please check #544.)

Now I'm thinking that there is room for improvement in current network API design, it's not just a problem of `global_step` variable. The purpose of `train.py` is just training. What the script need to know about the network should be following:

- how to initialize the network
- how to execute a training step
- how to execute a validation step

`train.py` need not to know how to run inference, how to compute a loss. Rather, it is better not to know.

Could you tell us your opnion about the idea above?

**Comment by Joeper214** [comment]
> I made a similar pull request a few days ago, but closed the pull request. (Please check #544.)

Oh. Sorry I didn't notice that. It seems like this PR makes no sense at all since I was only thinking of a temporary fix. I'll be closing this PR in that case.

### Regarding 
* how to initialize the network
* how to execute a training step
* how to execute a validation step

I agree as it will make train.py simpler and straightforward. 

However, I'm not sure how we can generate the gradients, summary, matrices and apply the loss function each step, and generate checkpoints in that case. Will it be in a separate API or executor?
